snuba/query/conditions.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Callable, Iterator
+
+from snuba.query.expressions import Expression, ExpressionContainer
+
+
+class Operator(Enum):
```

Comment:
Would it be simpler (on the internal query data structure side — not the parsing side) to just turn all of these infix operators into their functional equivalents, and remove the special casing for `Condition` types?

I'd think that this would be beneficial for any query processor to not have to handle both cases, and just expect everything to be functional. (I think the externally-facing query language should accept both.)

Reply:
That's a good question, I did not think about that assuming (probably incorrectly) that not all conditions could be expressed with functional equivalents.

I need to think about it for a second:
- yes it would make the structure simpler. 
- I need to verify whether query processing gains anything from explicit conditions.

Reply:
Turned Conditions into FunctionCall. This makes the conditions.py file simpler.
Right now it only states the valid function names, since the query processing code will have to depend  on those names to identify conditions, it is better not store these function names somewhere.

In the next step when I will introduce format and parsing, I am almost sure, I will reintroduce Some sort of Condition class. This time it will inherit from FunctionCall. This will allow us to format condition in the infix form (even when managed as functions) and it will simplify query processing (rely on class hierarchy instead of a string field name to identify the type of condition).
All this is not needed at this point, so I will bring it back later.